### PR TITLE
fix(components): [select] click label only focus in form

### LINKED ISSUE #9781: [Component] [form-item] form表单点击下拉框的label，触发了打开，再次点击却不能收起，点击时候可以聚焦，但不希望打开，参考ant框架
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.20`
- Element Plus Version: `2.2.16`
- Browser / OS: `99.0.4844.51`
- Build Tool: `Vite`

### Related Component

- `el-form-item`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuICA8ZWwtZm9ybSA6bW9kZWw9XCJmb3JtXCIgbGFiZWwtd2lkdGg9XCIxMjBweFwiPlxuICAgIDxlbC1mb3JtLWl0ZW0gbGFiZWw9XCJBY3Rpdml0eSBuYW1lXCI+XG4gICAgICA8ZWwtaW5wdXQgdi1tb2RlbD1cImZvcm0ubmFtZVwiIC8+XG4gICAgPC9lbC1mb3JtLWl0ZW0+XG4gICAgPGVsLWZvcm0taXRlbSBsYWJlbD1cIkFjdGl2aXR5IHpvbmVcIj5cbiAgICAgIDxlbC1zZWxlY3Qgdi1tb2RlbD1cImZvcm0ucmVnaW9uXCIgcGxhY2Vob2xkZXI9XCJwbGVhc2Ugc2VsZWN0IHlvdXIgem9uZVwiPlxuICAgICAgICA8ZWwtb3B0aW9uIGxhYmVsPVwiWm9uZSBvbmVcIiB2YWx1ZT1cInNoYW5naGFpXCIgLz5cbiAgICAgICAgPGVsLW9wdGlvbiBsYWJlbD1cIlpvbmUgdHdvXCIgdmFsdWU9XCJiZWlqaW5nXCIgLz5cbiAgICAgIDwvZWwtc2VsZWN0PlxuICAgIDwvZWwtZm9ybS1pdGVtPlxuICAgIDxlbC1mb3JtLWl0ZW0gbGFiZWw9XCJBY3Rpdml0eSB0aW1lXCI+XG4gICAgICA8ZWwtY29sIDpzcGFuPVwiMTFcIj5cbiAgICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgICAgdi1tb2RlbD1cImZvcm0uZGF0ZTFcIlxuICAgICAgICAgIHR5cGU9XCJkYXRlXCJcbiAgICAgICAgICBwbGFjZWhvbGRlcj1cIlBpY2sgYSBkYXRlXCJcbiAgICAgICAgICBzdHlsZT1cIndpZHRoOiAxMDAlXCJcbiAgICAgICAgLz5cbiAgICAgIDwvZWwtY29sPlxuICAgICAgPGVsLWNvbCA6c3Bhbj1cIjJcIiBjbGFzcz1cInRleHQtY2VudGVyXCI+XG4gICAgICAgIDxzcGFuIGNsYXNzPVwidGV4dC1ncmF5LTUwMFwiPi08L3NwYW4+XG4gICAgICA8L2VsLWNvbD5cbiAgICAgIDxlbC1jb2wgOnNwYW49XCIxMVwiPlxuICAgICAgICA8ZWwtdGltZS1waWNrZXJcbiAgICAgICAgICB2LW1vZGVsPVwiZm9ybS5kYXRlMlwiXG4gICAgICAgICAgcGxhY2Vob2xkZXI9XCJQaWNrIGEgdGltZVwiXG4gICAgICAgICAgc3R5bGU9XCJ3aWR0aDogMTAwJVwiXG4gICAgICAgIC8+XG4gICAgICA8L2VsLWNvbD5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgICA8ZWwtZm9ybS1pdGVtIGxhYmVsPVwiSW5zdGFudCBkZWxpdmVyeVwiPlxuICAgICAgPGVsLXN3aXRjaCB2LW1vZGVsPVwiZm9ybS5kZWxpdmVyeVwiIC8+XG4gICAgPC9lbC1mb3JtLWl0ZW0+XG4gICAgPGVsLWZvcm0taXRlbSBsYWJlbD1cIkFjdGl2aXR5IHR5cGVcIj5cbiAgICAgIDxlbC1jaGVja2JveC1ncm91cCB2LW1vZGVsPVwiZm9ybS50eXBlXCI+XG4gICAgICAgIDxlbC1jaGVja2JveCBsYWJlbD1cIk9ubGluZSBhY3Rpdml0aWVzXCIgbmFtZT1cInR5cGVcIiAvPlxuICAgICAgICA8ZWwtY2hlY2tib3ggbGFiZWw9XCJQcm9tb3Rpb24gYWN0aXZpdGllc1wiIG5hbWU9XCJ0eXBlXCIgLz5cbiAgICAgICAgPGVsLWNoZWNrYm94IGxhYmVsPVwiT2ZmbGluZSBhY3Rpdml0aWVzXCIgbmFtZT1cInR5cGVcIiAvPlxuICAgICAgICA8ZWwtY2hlY2tib3ggbGFiZWw9XCJTaW1wbGUgYnJhbmQgZXhwb3N1cmVcIiBuYW1lPVwidHlwZVwiIC8+XG4gICAgICA8L2VsLWNoZWNrYm94LWdyb3VwPlxuICAgIDwvZWwtZm9ybS1pdGVtPlxuICAgIDxlbC1mb3JtLWl0ZW0gbGFiZWw9XCJSZXNvdXJjZXNcIj5cbiAgICAgIDxlbC1yYWRpby1ncm91cCB2LW1vZGVsPVwiZm9ybS5yZXNvdXJjZVwiPlxuICAgICAgICA8ZWwtcmFkaW8gbGFiZWw9XCJTcG9uc29yXCIgLz5cbiAgICAgICAgPGVsLXJhZGlvIGxhYmVsPVwiVmVudWVcIiAvPlxuICAgICAgPC9lbC1yYWRpby1ncm91cD5cbiAgICA8L2VsLWZvcm0taXRlbT5cbiAgICA8ZWwtZm9ybS1pdGVtIGxhYmVsPVwiQWN0aXZpdHkgZm9ybVwiPlxuICAgICAgPGVsLWlucHV0IHYtbW9kZWw9XCJmb3JtLmRlc2NcIiB0eXBlPVwidGV4dGFyZWFcIiAvPlxuICAgIDwvZWwtZm9ybS1pdGVtPlxuICAgIDxlbC1mb3JtLWl0ZW0+XG4gICAgICA8ZWwtYnV0dG9uIHR5cGU9XCJwcmltYXJ5XCIgQGNsaWNrPVwib25TdWJtaXRcIj5DcmVhdGU8L2VsLWJ1dHRvbj5cbiAgICAgIDxlbC1idXR0b24+Q2FuY2VsPC9lbC1idXR0b24+XG4gICAgPC9lbC1mb3JtLWl0ZW0+XG4gIDwvZWwtZm9ybT5cbjwvdGVtcGxhdGU+XG5cbjxzY3JpcHQgbGFuZz1cInRzXCIgc2V0dXA+XG5pbXBvcnQgeyByZWFjdGl2ZSB9IGZyb20gJ3Z1ZSdcblxuLy8gZG8gbm90IHVzZSBzYW1lIG5hbWUgd2l0aCByZWZcbmNvbnN0IGZvcm0gPSByZWFjdGl2ZSh7XG4gIG5hbWU6ICcnLFxuICByZWdpb246ICcnLFxuICBkYXRlMTogJycsXG4gIGRhdGUyOiAnJyxcbiAgZGVsaXZlcnk6IGZhbHNlLFxuICB0eXBlOiBbXSxcbiAgcmVzb3VyY2U6ICcnLFxuICBkZXNjOiAnJyxcbn0pXG5cbmNvbnN0IG9uU3VibWl0ID0gKCkgPT4ge1xuICBjb25zb2xlLmxvZygnc3VibWl0IScpXG59XG48L3NjcmlwdD5cbiIsImltcG9ydF9tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHt9XG59IiwiX28iOnt9fQ==)

### Steps to reproduce

正常使用组件

### What is Expected?

点击下拉框的label，只聚焦不下拉，或者下拉再次点击可以收起

### What is actually happening?

点击label，打开下拉框，不能收起了

### Additional comments

_(empty)_

### PR DESCRIPTION
**Pull request by chenxch**
close #9781 

- [ ] Make sure you follow contributing guide [English](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.en-US.md) | ([中文](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.zh-CN.md) | [Español](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.es.md) | [Français](https://github.com/element-plus/element-plus/blob/master/.github/CONTRIBUTING.fr-FR.md)).
- [ ] Make sure you are merging your commits to `dev` branch.
- [ ] Add some descriptions and refer to relative issues for your PR.

before
select1

after
select2

### CODE REVIEW

**Comment by chenxch** [comment]
At present, it is judged by pointerType. If there are hidden problems, please help to confirm together.

**Comment by chenxch** [comment]
At present, it is judged by pointerType. If there are hidden problems, please help to confirm together.
cc @element-plus/backers

File: packages/components/select/src/select.vue [hunk 1]
```diff
@@ -3,7 +3,7 @@
     ref="selectWrapper"
     v-click-outside:[popperPaneRef]="handleClose"
     :class="wrapperKls"
-    @click.stop="toggleMenu"
+    @click.stop="(e) => toggleMenu(e)"
```
Comment (sxzz) [inline packages/components/select/src/select.vue:6]: The event will be passed by default. Maybe we can revert this change.

**Comment by sxzz** [comment]
@chenxch [`pointerType`](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerType) is only supported by Safari >= 13 (Our compat table is Safari >= 12)

**Comment by sxzz** [comment]
Perhaps, we can use `mouseEnter` to judge it.

**Comment by chenxch** [comment]
> @chenxch [`pointerType`](https://developer.mozilla.org/en-US/docs/Web/API/PointerEvent/pointerType) is only supported by Safari >= 13 (Our compat table is Safari >= 12)

It really needs to be replaced with a more compatible solution.

**Comment by chenxch** [comment]
> Perhaps, we can use `mouseEnter` to judge it.

Great idea, I'll try it out.

File: packages/components/select/src/select.vue (hunk 1 above)
Comment (chenxch) [reply packages/components/select/src/select.vue:6]: ok.

**Comment by fenghua123** [comment]
> 🧪 Playground Preview: https://element-plus.run/?pr=9798 Please comment the example via this playground if needed.

That is what I need

**Comment by chenxch** [comment]
@sxzz PTAL

**Review by sxzz — APPROVED** [review]

**Comment by robert-wloch-iits** [comment]
Please consider rolling back that change. It brings poor UX and inconsistent UX to forms.
Consider "Basic Form" demo on the Form documentation page: [https://element-plus.org/en-US/component/form.html#basic-form](https://element-plus.org/en-US/component/form.html#basic-form)

Clicking the label of a form item should really put the focus into the field or trigger actions. It is done with these labels/fields:

- Activity name -> focusses the text input *as expected by users*
- Instant delivery -> toggles the switch *as expected by users* (you don't have to click the switch itself)
- Activity form -> focusses the text input *as expected by users*

On the contrary: What is the use for screen readers of now having to do a mouse enter before the select can be opened via click?

Regarding inconsistency of clicking labels (examples again from Form documentation -> Basic Form):
- Activity zone -> blue focus border is rendered but nothing opens anymore
- Activity time -> no blue border and nothing opens anymore
- Activity type -> no blue border on first checkbox. User has to use tab to get to first checkbox. BUT: Clicking the label of each checkbox toggles the checkboxes *as expected by users* (you don't have to click the checkbox itself)
- Resources -> no blue border on first radio button. User has to use tab to get to first radio button. BUT: Clicking the label of each radio button selects the radio button *as expected by users* (you don't have to click the circular radio button itself)

**Comment by jw-foss** [comment]
@chenxch I do think @robert-wloch-iits make the point, why were having changes like these? Would you mind fill me in a little bit? Thanks.

**Comment by chenxch** [comment]
> @chenxch I do think @robert-wloch-iits make the point, why were having changes like these? Would you mind fill me in a little bit? Thanks.

I made this adjustment according to the W3C specification.

**Comment by sxzz** [comment]
issue https://github.com/element-plus/element-plus/issues/9781

It's the reason for respecting the behavior of the native web. 

See the [demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHRlbXBsYXRlPlxuIDxmb3JtIGFjdGlvbj1cIi9hY3Rpb25fcGFnZS5waHBcIj5cbsKgIDxpbnB1dCB0eXBlPVwicmFkaW9cIiBpZD1cImh0bWxcIiBuYW1lPVwiZmF2X2xhbmd1YWdlXCIgdmFsdWU9XCJIVE1MXCI+XG7CoCA8bGFiZWwgZm9yPVwiaHRtbFwiPkhUTUw8L2xhYmVsPjxicj5cbsKgIDxpbnB1dCB0eXBlPVwicmFkaW9cIiBpZD1cImNzc1wiIG5hbWU9XCJmYXZfbGFuZ3VhZ2VcIiB2YWx1ZT1cIkNTU1wiPlxuwqAgPGxhYmVsIGZvcj1cImNzc1wiPkNTUzwvbGFiZWw+PGJyPlxuwqAgPGxhYmVsIGZvcj1cImphdmFzY3JpcHRcIj5KYXZhU2NyaXB0PC9sYWJlbD48YnI+PGJyPlxuwqAgPHNlbGVjdCB0eXBlPVwidGV4dFwiIGlkPVwiamF2YXNjcmlwdFwiIG5hbWU9XCJmYXZfbGFuZ3VhZ2VcIiBzdHlsZT1cIndpZHRoOjIwMHB4XCI+XG4gIFx0PG9wdGlvbiB2YWx1ZT1cIlwiPi0tUGxlYXNlIGNob29zZSBhbiBvcHRpb24tLTwvb3B0aW9uPlxuICAgIDxvcHRpb24gdmFsdWU9XCJkb2dcIj5Eb2c8L29wdGlvbj5cbiAgICA8b3B0aW9uIHZhbHVlPVwiY2F0XCI+Q2F0PC9vcHRpb24+XG4gICAgPG9wdGlvbiB2YWx1ZT1cImhhbXN0ZXJcIj5IYW1zdGVyPC9vcHRpb24+XG4gICAgPG9wdGlvbiB2YWx1ZT1cInBhcnJvdFwiPlBhcnJvdDwvb3B0aW9uPlxuICAgIDxvcHRpb24gdmFsdWU9XCJzcGlkZXJcIj5TcGlkZXI8L29wdGlvbj5cbiAgICA8b3B0aW9uIHZhbHVlPVwiZ29sZGZpc2hcIj5Hb2xkZmlzaDwvb3B0aW9uPlxuICA8L3NlbGVjdD5cbiAgPGlucHV0IHR5cGU9XCJzdWJtaXRcIiB2YWx1ZT1cIlN1Ym1pdFwiPlxuPC9mb3JtPlxuPC90ZW1wbGF0ZT5cblxuPHNjcmlwdCBsYW5nPVwidHNcIiBzZXR1cD5cblxuPC9zY3JpcHQ+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=). When clicking "JavaScript" label, it will only trigger focus on select, instead of expanding it.

**Comment by jw-foss** [comment]
I did some research on those popular design systems for example [atlanssia design](https://atlassian.design/components/form/examples#select), click on the label it only focuses on the input element nested inside the select wrapper, and it **DOES NOT** open the dropdown automatically. 🤔 I think this matches the W3C principle since the native select element combined with label will also behave this way. You may also refer to [this](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/select).

**Comment by robert-wloch-iits** [comment]
I see, thanks for the references and clarifications!

**Comment by AuSjukwa** [comment]
@chenxch Hello, I still can reproduce the problem on the website doc demo and in version 2.7.3. It's there a rolling back?

**Comment by YunYouJun** [comment]
image

I did reproduce it, but strangely enough the last time we modified it was two years ago.
Can you have a look? @chenxch

**Pull request merged**